internal/services/mongodb/snapshot.go

```diff
@@ -0,0 +1,212 @@
+package mongodb
+
+import (
+	"context"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/customdiff"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	mongodb "github.com/scaleway/scaleway-sdk-go/api/mongodb/v1alpha1"
+	"github.com/scaleway/scaleway-sdk-go/scw"
+	"github.com/scaleway/terraform-provider-scaleway/v2/internal/locality"
+	"github.com/scaleway/terraform-provider-scaleway/v2/internal/locality/regional"
+	"github.com/scaleway/terraform-provider-scaleway/v2/internal/locality/zonal"
+	"github.com/scaleway/terraform-provider-scaleway/v2/internal/types"
+	"github.com/scaleway/terraform-provider-scaleway/v2/internal/verify"
+)
+
+func ResourceSnapshot() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: ResourceSnapshotCreate,
+		ReadContext:   ResourceSnapshotRead,
+		UpdateContext: ResourceSnapshotUpdate,
+		DeleteContext: ResourceSnapshotDelete,
+		Timeouts: &schema.ResourceTimeout{
+			Create:  schema.DefaultTimeout(defaultMongodbSnapshotTimeout),
+			Update:  schema.DefaultTimeout(defaultMongodbSnapshotTimeout),
+			Delete:  schema.DefaultTimeout(defaultMongodbSnapshotTimeout),
+			Default: schema.DefaultTimeout(defaultMongodbSnapshotTimeout),
+		},
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+		SchemaVersion: 0,
+		Schema: map[string]*schema.Schema{
+			"id": {
```

Comment:
Might be confusing with the SetId method. Why is this required?

Reply:
You are right